nodes/src/nodes/tool_filesystem/IInstance.py

```
@@ -0,0 +1,328 @@
+                    'default': 'utf-8',
+                },
+            },
+            'additionalProperties': False,
+        },
+        description=('Read a file from the account file store and return its contents as a decoded string. Required: "path" (relative path). Optional: "encoding" (default "utf-8"). Returns: {path, content, size} where size is the byte length before decoding.'),
+    )
+    def read_file(self, args):
+        args = _require_dict(args)
+        self._check_ready()
+        if not self.IGlobal.allow_read:
+            raise ValueError('read access is not enabled for this filesystem tool')
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)